### PR TITLE
[REFACTOR][Relay] Migrate Id ObjectRef to not-null

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
Refer #5318

@tqchen, @jroesch, @zhiics, @junrushao1994   : Please help review, Thanks!

### CODE REVIEW

File: include/tvm/relay/expr.h [hunk 1]
```diff
@@ -152,6 +152,9 @@ class Var;
 /*! \brief Container for Var */
 class VarNode : public ExprNode {
  public:
+  VarNode(Id vid, Type type_annotation) : vid(vid), type_annotation(type_annotation) {}
+  VarNode() : vid("") {}
```
Comment (tqchen) [inline include/tvm/relay/expr.h:156]: The not-null that involves composite objects are more complicated and this is not necessarily the best approach, let us wait a bit before we find a better solution

**Comment by ANSHUMAN87** [comment]
This would be a baseline reference for other Relay nodes.
I have one query here. As you see i have to add 1 empty/default constructor here to resolve compile error i faced in [TVM_REGISTER_NODE_TYPE(VarNode)](https://github.com/apache/incubator-tvm/blob/6ae439c8c58dd0118d2f2c5d1c4bcb650df47104/src/relay/ir/expr.cc#L91).

Please suggest the best way to cover this! Thanks!

File: include/tvm/relay/expr.h (hunk 1 above)
Comment (ANSHUMAN87) [reply include/tvm/relay/expr.h:156]: > The not-null that involves composite objects are more complicated and this is not necessarily the best approach, let us wait a bit before we find a better solution

@tqchen : Thanks! I also felt that this is not best way. We might have to change Node registration framework i believe. Will wait for your future guidance!

**Comment by junrushao1994** [comment]
@tqchen Seems that some changes in my WIP PR have affected this PR https://ci.tvm.ai/blue/organizations/jenkins/tvm/detail/PR-5748/1/pipeline. We probably should "make clean" before build wasm

**Comment by tqchen** [comment]
@junrushao1994 good pt, can you send another PR?

**Pull request closed without merging**